conf/template/ICUBConfig.cmake.in

```diff
@@ -0,0 +1,131 @@
+# Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+#
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+
+
+include(CMakeDependentOption)
+
+message(STATUS "@ICUB_MESSAGE@")
+
+
+# Version
+set(ICUB_VERSION_MAJOR "@ICUB_VERSION_MAJOR@")
+set(ICUB_VERSION_MINOR "@ICUB_VERSION_MINOR@")
+set(ICUB_VERSION_PATCH "@ICUB_VERSION_PATCH@")
+set(ICUB_VERSION_TWEAK "@ICUBP_VERSION_TWEAK@")
+set(ICUB_VERSION "@ICUB_VERSION@")
+set(ICUB_VERSION_SHORT "@ICUB_VERSION_SHORT@")
+
+# Build type and flags
+set(ICUB_IS_SHARED_LIBRARY @BUILD_SHARED_LIBS@)
+
+set(ICUB_C_COMPILER_ID "@CMAKE_C_COMPILER_ID@")
+set(ICUB_C_COMPILER_VERSION "@CMAKE_C_COMPILER_VERSION@")
+set(ICUB_C_FLAGS "@CMAKE_C_FLAGS@")
+
+set(ICUB_CXX_COMPILER_ID "@CMAKE_CXX_COMPILER_ID@")
+set(ICUB_CXX_COMPILER_VERSION "@CMAKE_CXX_COMPILER_VERSION@")
+set(ICUB_CXX_FLAGS "@CMAKE_CXX_FLAGS@")
+
+
+@PACKAGE_INIT@
+
+
+set(ICUB_MODULE_PATH "@ICUB_MODULE_PATH@" CACHE INTERNAL "iCub cmake scripts directory")
+set(ICUB_LIBRARIES "@ICUB_TARGETS@" CACHE INTERNAL "List of iCub libraries")
+set(ICUB_LINK_FLAGS "@ICUB_LINK_FLAGS@" CACHE INTERNAL "List of iCub linker options")
+set(ICUB_LINK_DIRECTORIES "@ICUB_LINK_DIRECTORIES@")
+set(ICUB_INSTALL_PREFIX "@CMAKE_INSTALL_PREFIX@")
+
+set(ICUB_PLUGIN_MANIFESTS_INSTALL_DIR "@ICUB_PLUGIN_MANIFESTS_INSTALL_DIR@")
+set(ICUB_MODULES_INSTALL_DIR "@ICUB_MODULES_INSTALL_DIR@")
+set(ICUB_APPLICATIONS_INSTALL_DIR "@ICUB_APPLICATIONS_INSTALL_DIR@")
+set(ICUB_TEMPLATES_INSTALL_DIR "@ICUB_TEMPLATES_INSTALL_DIR@")
+set(ICUB_APPLICATIONS_TEMPLATES_INSTALL_DIR "@ICUB_APPLICATIONS_TEMPLATES_INSTALL_DIR@")
+set(ICUB_MODULES_TEMPLATES_INSTALL_DIR "@ICUB_MODULES_TEMPLATES_INSTALL_DIR@")
+set(ICUB_CONTEXTS_INSTALL_DIR "@ICUB_CONTEXTS_INSTALL_DIR@")
```

Comment:
There is a specific reason for exporting so many variables? What are the use cases for this variables?  To avoid problems in the future I suggest to export the minimum number of variables.

Reply:
I exported them for backward compatibility, since they were already exported, see https://github.com/robotology/icub-main/blob/master/conf/template/icub-config-install.cmake.in#L20